client/src/pages/challenges.js

```diff
@@ -0,0 +1,97 @@
+import { Component } from 'preact'
+import config from '../config'
+import 'linkstate/polyfill'
+import withStyles from '../components/jss'
+
+import { getChallenges } from '../api/challenges'
+
+export default withStyles({
+  frame: {
+    marginBottom: '15px',
+    paddingBottom: '10px'
+  },
+  divider: {
+    margin: '10px',
+    width: '80%'
+  }
+}, class Profile extends Component {
+  state = {
+    problems: []
+  }
+
+  componentDidMount () {
+    document.title = 'Challenges' + config.ctfTitle
+
+    getChallenges()
+      .then(problems => {
+        this.setState({
+          problems
+        })
+      })
+  }
+
+  render ({ classes }, { problems }) {
+    return (
```

Comment:
Refactoring this into multiple components would be better.

Reply:
I'll make an issue for this